tests/stdlib/thttpclient_standalone.nim

```diff
@@ -0,0 +1,32 @@
+discard """
+  cmd: "nim c --threads:on $file"
+  output: "OK"
+"""
+
+import asynchttpserver, httpclient, asyncdispatch
+
+block: # issue #16436
+  proc startServer() {.async.} =
+    proc cb(req: Request) {.async.} =
+      let headers = { "Content-length": "15"} # Provide invalid content-length
+      await req.respond(Http200, "Hello World", headers.newHttpHeaders())
+
+    var server = newAsyncHttpServer()
+    await server.serve(Port(5555), cb)
```

Comment:
causes flaky tests, see https://github.com/nim-lang/Nim/pull/16678#discussion_r555450480 and how that PR solves it; let's wait until https://github.com/nim-lang/Nim/pull/16678 is merged so you can use `let port = getLocalAddr(server.getSocket.getFd, AF_INET)[1]`

Reply:
Why does this cause flaky tests? Is it because the port is too low? Maybe a higher port number can just be used?

Reply:
see https://github.com/nim-lang/Nim/pull/14327 which has all the details. Multiple tests can potentially access the same port and when they run in parallel, you get flaky tests that trigger based on unrelated conditions (eg simply reordering a file etc).

in future PR we can make it easier for users to get the port, in the meantime we can use `let port = getLocalAddr(server.getSocket.getFd, AF_INET)[1]`